rpc/params/legacy.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package params
+
+import (
+	"time"
+
+	"github.com/juju/names/v6"
+
+	"github.com/juju/juju/core/life"
+	"github.com/juju/juju/core/model"
+	"github.com/juju/juju/core/semversion"
+)
+
+// ApproximateUserTagFromQualifier creates a valid user tag
+// from the supplied qualifier. A qualifier does not contain
+// all the same characters that a user tag can have.
```

Comment:
Like?

Reply:
A username can contain "@" and "+" which a qualifier cannot. Updated comment.